frontend/src/pages/addCategory.js

```diff
@@ -0,0 +1,55 @@
+import React, { useEffect, useState } from "react";
+import Container from "react-bootstrap/Container";
+import CategoryForm from "../components/categoryForm";
+import { useNavigate } from "react-router-dom";
+import { addCategory } from "../services/adminService";
+import Row from "react-bootstrap/esm/Row";
+import Col from "react-bootstrap/esm/Col";
+
+const AddCategory = () => {
+  const navigate = useNavigate();
+
+  const [state, setState] = useState(null);
+  const [error, setError] = useState();
+  const [success, setSuccess] = useState(false);
+
+  const handleSave = async () => {
+    addCategory(state)
+      .then(() => {
+        setSuccess(true);
+      })
+      .catch(({ response }) => {
+        setError(response.data);
+      })
+      .finally(() => {});
+  };
+
+  const handleChange = ({ currentTarget: input }) => {
+    const name = input.id;
+    const value = input.value;
+
+    setState({ ...state, [name]: value });
```

Comment:
```suggestion
    setState({ ...state, [input.id]: input.value });
```
dont create a variable unless it will be used more than 2 times coz it will allocate memory